engine/table/src/main/java/io/deephaven/engine/table/impl/by/typed/TypedHasherFactory.java

```diff
@@ -0,0 +1,606 @@
+     */
+    public static <T> T make(HasherConfig<T> hasherConfig, ColumnSource<?>[] tableKeySources, int tableSize,
+            double maximumLoadFactor, double targetLoadFactor) {
+        final ChunkType[] chunkTypes =
+                Arrays.stream(tableKeySources).map(ColumnSource::getChunkType).toArray(ChunkType[]::new);
+        if (USE_PREGENERATED_HASHERS) {
+            if (hasherConfig.baseClass.equals(StaticChunkedOperatorAggregationStateManagerTypedBase.class)) {
+                // noinspection unchecked
+                T pregeneratedHasher = (T) io.deephaven.engine.table.impl.by.typed.staticagg.gen.TypedHashDispatcher
+                        .dispatch(tableKeySources, tableSize, maximumLoadFactor, targetLoadFactor);
+                if (pregeneratedHasher != null) {
+                    return pregeneratedHasher;
+                }
+            } else if (hasherConfig.baseClass
+                    .equals(IncrementalChunkedOperatorAggregationStateManagerTypedBase.class)) {
+                // noinspection unchecked
+                T pregeneratedHasher = (T) io.deephaven.engine.table.impl.by.typed.incagg.gen.TypedHashDispatcher
+                        .dispatch(tableKeySources, tableSize, maximumLoadFactor, targetLoadFactor);
+                if (pregeneratedHasher != null) {
+                    return pregeneratedHasher;
+                }
+            }
+        }
+
+        final String className = hasherName(chunkTypes);
+
+        JavaFile javaFile = generateHasher(hasherConfig, chunkTypes, className, Optional.of(Modifier.PUBLIC));
+
+        String[] javaStrings = javaFile.toString().split("\n");
+        final String javaString =
+                Arrays.stream(javaStrings).filter(s -> !s.startsWith("package ")).collect(Collectors.joining("\n"));
+
+        final Class<?> clazz = CompilerTools.compile(className, javaString, packageName(hasherConfig.packageMiddle));
+        if (!hasherConfig.baseClass.isAssignableFrom(clazz)) {
+            throw new IllegalStateException("Generated class is not a " + hasherConfig.baseClass.getCanonicalName());
+        }
+
+        final Class<? extends T> castedClass = (Class<? extends T>) clazz;
+
+        T retVal;
+        try {
+            final Constructor<? extends T> constructor1 =
+                    castedClass.getDeclaredConstructor(ColumnSource[].class, int.class, double.class, double.class);
+            retVal = constructor1.newInstance(tableKeySources, tableSize, maximumLoadFactor, targetLoadFactor);
+        } catch (InstantiationException | IllegalAccessException | InvocationTargetException
+                | NoSuchMethodException e) {
+            throw new UncheckedDeephavenException("Could not instantiate " + castedClass.getCanonicalName(), e);
+        }
+        return retVal;
+    }
+
+    @NotNull
+    public static String packageName(String packageMiddle) {
+        return "io.deephaven.engine.table.impl.by.typed." + packageMiddle + ".gen";
```

Comment:
You'll need more structure for joins.

Reply:
I will burn that bridge when/if we decide to get there.